### PR TITLE
Added logic for checkign gcp nats on instances, will skip hosts without a NAT ip

### PR DESCRIPTION
**Pull request by maswertyDT**
This code should change how the Nat IP checking works, it should skip hosts that do not have a public IP attached to the VM instances in GCP

### CODE REVIEW

**Comment by maswertyDT** [comment]
@maximillianbrain1 Could you review this PR when you get a chance? thanks!!

File: pkg/gcloud/gcloud.go [hunk 1]
```diff
@@ -61,14 +62,18 @@ func (g *gCloudSvc) Instances(serversMap map[string]server.Server) error {
...
-			newServer.Tags = make(map[string]string)
-			newServer.Name = instance.Name
-			newServer.Address = instance.NetworkInterfaces[0].AccessConfigs[0].NatIP
-			for index, key := range instance.Tags.Items {
-				newServer.Tags[strconv.FormatInt(int64(index), 10)] = key
+			if len(instance.NetworkInterfaces) > 0 {
+				newServer := server.Server{}
+				newServer.Tags = make(map[string]string)
+				newServer.Name = instance.Name
+				newServer.Address = instance.NetworkInterfaces[0].AccessConfigs[0].NatIP
```
Comment (maximillianbrain1) [inline pkg/gcloud/gcloud.go:69]: What happens if an Instance has a Network Interface that does not have an AccessConfig? We should iterate through all Network Interfaces and all AccessConfig Structs.

**Review by maximillianbrain1 — CHANGES_REQUESTED** [review]
Overall it looks pretty good. I would iterate through all NetworkInterfaces and AccessConfig Structs for a NatIP. Don't forget to create tests for the changes in gcloud_test.go!

**Comment by maswertyDT** [comment]
@maximillianbrain1 I believe I've added the changes you've requested, I think the code now checks for the variable that contains the network interfaces & AccessConfig Structs

File: pkg/gcloud/gcloud.go [hunk 2]
```diff
@@ -61,14 +62,22 @@ func (g *gCloudSvc) Instances(serversMap map[string]server.Server) error {
...
```
Comment (maximillianbrain1) [inline pkg/gcloud/gcloud.go:69]: What happens if AccessConfigs is an empty slice?

File: pkg/gcloud/gcloud.go [hunk 3]
```diff
@@ -61,14 +62,22 @@ func (g *gCloudSvc) Instances(serversMap map[string]server.Server) error {
...
+				if len(newServer.Address) > 0 {
```
Comment (maximillianbrain1) [inline pkg/gcloud/gcloud.go:70]: How do we know that this `Address` is the NatIP?

File: pkg/gcloud/gcloud_test.go [hunk 4]
```diff
@@ -67,6 +67,14 @@ func TestGetInstances(t *testing.T) {
 				},
 			},
 		},
+		{
```
Comment (maximillianbrain1) [inline pkg/gcloud/gcloud_test.go:70]: What are you trying to test here? How is this different from Instance 2?

**Review by maximillianbrain1 — CHANGES_REQUESTED** [review]
Getting closer, but there is still a chance for the code to fail if there is no AccessConfig for a given Network Interface.

Let me know if you having difficulty and want to pair on it!

File: pkg/gcloud/gcloud.go (hunk 2 above)
Comment (maswertyDT) [reply pkg/gcloud/gcloud.go:69]: Would love to pair on this as I've tried a few different ways to approach this, however, the only way I know to address the conditional statement is to check if newServer.Address has any characters/strings associated to it (>0), if you have any ideas, I'd love to work with you on that!!

File: pkg/gcloud/gcloud_test.go (hunk 4 above)
Comment (maswertyDT) [reply pkg/gcloud/gcloud_test.go:70]: My attempt here was to simulate another instance but no NatIP

**Comment by maswertyDT** [comment]
@maximillianbrain1 when you get a moment, could you review this PR? I was able to fix this with the latest commit. Thanks!! :)

**Review by maximillianbrain1 — APPROVED** [review]

**Pull request merged**